### PR TITLE
Databases major refactoring

### PR DESCRIPTION
**Pull request by JIMENOFONSECA**
This should do for #2325 and #1328 and #2366 

The next modifications are included:
1. all databases are renamed to simpler and more intuitive names.
2. the LCA-infrastructure database is moved to the system's database. 
3. the tehcnical_systems database is renamed to air-conditioning systems.
3. the systems_controls database is erased and it is replaced for 4 parameters in the input database of air-conditioning_systems which reflect the start and end of the heating seasons. I added some assertion rules so there is no overlap ( no supported)

To test: try to run all the cea workflow, with the CEA tool. You will need to start form the datahelper 
tool.

@Jack-Hawthorne and @daren-thomas mind checking this please?

### CODE REVIEW

**Comment by JIMENOFONSECA** [comment]
@daren-thomas @Jack-Hawthorne the dodo pas locally but the jenkins seems to not like this PR (or to be offline)

**Comment by gabriel-happle** [comment]
@JIMENOFONSECA the unit tests are failing. other tests are ok.
```
.  run_unit_tests
FF..............EE
======================================================================
ERROR: test_mixed_use_archetype_values (test_schedules.TestBuildingPreprocessing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gabriel\documents\cityenergyanalyst\cityenergyanalyst\cea\tests\test_schedules.py", line 52, in test_mixed_use_archetype_values
    list_uses=['SERVERROOM', 'PARKING']),([0.5, 0.2], [0.0, 0.0], [0.5, 0.2], [0.95, 0.9200000000000002]))
  File "c:\users\gabriel\documents\cityenergyanalyst\cityenergyanalyst\cea\datamanagement\data_helper.py", line 394, in correct_archetype_areas
    Hs_ag = calc_average(Hs_ag, indexed_DB['Hs_ag'][current_use_code], prop_architecture_df[use][building])
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\frame.py", line 2688, in __getitem__
    return self._getitem_column(key)
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\frame.py", line 2695, in _getitem_column
    return self._get_item_cache(key)
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\generic.py", line 2489, in _get_item_cache
    values = self._data.get(item)
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\internals.py", line 4115, in get
    loc = self.items.get_loc(item)
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\indexes\base.py", line 3080, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 140, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 162, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1492, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1500, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'Hs_ag'

======================================================================
ERROR: test_mixed_use_schedules (test_schedules.TestScheduleCreation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gabriel\documents\cityenergyanalyst\cityenergyanalyst\cea\tests\test_schedules.py", line 61, in test_mixed_use_schedules
    building_properties = BuildingProperties(locator, False)
  File "c:\users\gabriel\documents\cityenergyanalyst\cityenergyanalyst\cea\demand\building_properties.py", line 77, in __init__
    prop_HVAC_result = get_properties_technical_systems(locator, prop_hvac).set_index('Name')
  File "c:\users\gabriel\documents\cityenergyanalyst\cityenergyanalyst\cea\demand\building_properties.py", line 774, in get_properties_technical_systems
    result = df_emission_heating[fields_emission_heating].merge(df_emission_cooling[fields_emission_cooling],
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\frame.py", line 2682, in __getitem__
    return self._getitem_array(key)
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\frame.py", line 2726, in _getitem_array
    indexer = self.loc._convert_to_indexer(key, axis=1)
  File "C:\Users\Gabriel\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\indexing.py", line 1327, in _convert_to_indexer
    .format(mask=objarr[mask]))
KeyError: "['heat_starts' 'heat_ends' 'cool_starts' 'cool_ends'] not in index"

======================================================================
FAIL: test_calc_thermal_loads (test_calc_thermal_loads.TestCalcThermalLoads)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gabriel\documents\cityenergyanalyst\cityenergyanalyst\cea\tests\test_calc_thermal_loads.py", line 77, in test_calc_thermal_loads
    column, values[i], df[column].sum()), places=3)
AssertionError: Sum of column E_sys_kWh differs, 397797.926000 != 400912.728000

======================================================================
FAIL: test_calc_thermal_loads_other_buildings (test_calc_thermal_loads.TestCalcThermalLoads)
Test some other buildings just to make sure we have the proper data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\gabriel\documents\cityenergyanalyst\cityenergyanalyst\cea\tests\test_calc_thermal_loads.py", line 99, in test_calc_thermal_loads_other_buildings
    places=3)
AssertionError: qhs_sys_kwh for B01 should be: 75227.90100, was 95850.97400

----------------------------------------------------------------------
Ran 18 tests in 37.263s

FAILED (failures=2, errors=2)
########################################
TaskFailed - taskid:run_unit_tests
Python Task failed: '<function run_unit_tests at 0x0000000010E94A58>' returned False
```

**Comment by JIMENOFONSECA** [comment]
Weird, I am not able to replicate...

**Comment by JIMENOFONSECA** [comment]
Update from Gabriel. It works now, the error was in the open case study which needed to be upgraded

File: cea/inputlocator.py [hunk 1]
```diff
@@ -652,11 +638,18 @@ def get_site_polygon(self):
         self.check_cpg(shapefile_path)
         return shapefile_path
 
-    def get_district_geometry(self):
-        """scenario/inputs/building-geometry/district.shp"""
-        shapefile_path = os.path.join(self.get_building_geometry_folder(), 'district.shp')
-        self.check_cpg(shapefile_path)
-        return shapefile_path
+    def get_surrounding_geometry(self):
```
Comment (daren-thomas) [inline cea/inputlocator.py]: this needs to be renamed to `get_surroundings_geometry`

**Comment by daren-thomas** [comment]
region "SG" has supply-systems.xls with the worksheet called "ALL_IN_ONE", but demand script expects "ALL_IN_ONE_SYSTEMS". are the rest of the worksheets ok?

**Comment by daren-thomas** [comment]
```
File "c:\users\daren-thomas\documents\cityenergyanalyst\cityenergyanalyst\cea\technologies\solar\photovoltaic.py", line 798, in main
    aggregated_hourly_results_df = aggregated_hourly_results_df.set_index('Date')
  File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\frame.py", line 3909, in set_index
    level = frame[col]._values
  File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\frame.py", line 2688, in __getitem__
    return self._getitem_column(key)
  File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\frame.py", line 2695, in _getitem_column
    return self._get_item_cache(key)
  File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\generic.py", line 2489, in _get_item_cache
    values = self._data.get(item)
  File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\internals.py", line 4115, in get
    loc = self.items.get_loc(item)
  File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pandas\core\indexes\base.py", line 3080, in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
  File "pandas/_libs/index.pyx", line 140, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 162, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1492, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1500, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'Date'
```

**Comment by daren-thomas** [comment]
Trying to create a new scenario with the GUI didn't work - using `yarn dev`, I was able to see the following error message:

```
2019-11-07 16:13:48,032] ERROR in app: Exception on /api/project/scenario/ [POST]
  Traceback (most recent call last):
    File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
      rv = self.dispatch_request()
    File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1799, in dispatch_request
      return self.view_functions[rule.endpoint](**req.view_args)
    File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask_restplus\api.py", line 325, in wrapper
      resp = resource(*args, **kwargs)
    File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\views.py", line 88, in view
      return self.dispatch_request(*args, **kwargs)
    File "C:\Users\daren-thomas\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask_restplus\resource.py", line 44, in dispatch_request
      resp = meth(*args, **kwargs)
    File "c:\users\daren-thomas\documents\cityenergyanalyst\cityenergyanalyst\cea\interfaces\dashboard\api\project.py", line 179, in post
      cea.api.district_helper(config)
  AttributeError: 'module' object has no attribute 'district_helper'
```

File: cea/datamanagement/surroundings_helper.py [hunk 2]
```diff
@@ -197,10 +197,10 @@ def geometry_extractor_osm(locator, config):
...
 
     # local variables:
-    buffer_m = config.district_helper.buffer
-    buildings_height = config.district_helper.height_ag
-    buildings_floors = config.district_helper.floors_ag
-    shapefile_out_path = locator.get_district_geometry()
+    buffer_m = config.surroundings_helper.buffer
+    buildings_height = config.surroundings_helper.height_ag
+    buildings_floors = config.surroundings_helper.floors_ag
+    shapefile_out_path = locator.get_surrounding_geometry()
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply cea/datamanagement/surroundings_helper.py]: fixed.

File: cea/inputlocator.py (hunk 1 above)
Comment (daren-thomas) [reply cea/inputlocator.py]: fixed

File: cea/demand/control_heating_cooling_systems.py [hunk 3]
```diff
@@ -240,17 +240,17 @@ def heating_system_is_active(tsd, t):
 
 def convert_date_to_hour(date):
     """
-    converts date in 'MM-DD' format into hour of the year (first hour of the day)
-    i.e. '01-01' results in 0
+    converts date in 'DD|MM' format into hour of the year (first hour of the day)
```
[the remark this answers is not in the document]
Comment (JIMENOFONSECA) [reply cea/demand/control_heating_cooling_systems.py]: sure, so now the inputs of the archetypes and input databases (airconditioning_systems.dbf) include the start and end of the heating and cooling season in this format.
The possible formats are:

1. DD|MM
2. DD-MM
3. DD/MM

The problem with using  2 or 3 is that once we read the files in excel, it is really easy for excel to transform them automatically into dates. Which leads to unwanted behavior.

Option 1. is both read by the database schema as string (intended) and also presents a separation between day and month.

File: cea/demand/control_heating_cooling_systems.py [hunk 4]
```diff
@@ -269,8 +269,8 @@ def is_heating_season(t, bpr):
 
     if bpr.hvac['has-heating-season']:
 
-        heating_start = convert_date_to_hour(bpr.hvac['heating-season-start'])
-        heating_end = convert_date_to_hour(bpr.hvac['heating-season-end']) + 23  # end at the last hour of the day
+        heating_start = convert_date_to_hour(bpr.hvac['heat_starts'])
```
[the remark this answers is not in the document]
Comment (JIMENOFONSECA) [reply cea/demand/control_heating_cooling_systems.py]: yes. the reason is that the input database air-conditioning.dbf does not accept long column names. So the shortest possible was 'heat_starts'

File: cea/demand/control_heating_cooling_systems.py [hunk 5]
```diff
@@ -269,8 +269,8 @@ def is_heating_season(t, bpr):
...
+        heating_end = convert_date_to_hour(bpr.hvac['heat_ends']) + 23  # end at the last hour of the day
```
[the remark this answers is not in the document]
Comment (JIMENOFONSECA) [reply cea/demand/control_heating_cooling_systems.py]: same as on top...

File: cea/demand/control_heating_cooling_systems.py [hunk 6]
```diff
@@ -307,8 +307,8 @@ def is_cooling_season(t, bpr):
 
     if bpr.hvac['has-cooling-season']:
 
-        cooling_start = convert_date_to_hour(bpr.hvac['cooling-season-start'])
-        cooling_end = convert_date_to_hour(bpr.hvac['cooling-season-end']) + 23  # end at the last hour of the day
+        cooling_start = convert_date_to_hour(bpr.hvac['cool_starts'])
```
[the remark this answers is not in the document]
Comment (JIMENOFONSECA) [reply cea/demand/control_heating_cooling_systems.py]: see above

File: cea/demand/control_heating_cooling_systems.py [hunk 7]
```diff
@@ -307,8 +307,8 @@ def is_cooling_season(t, bpr):
...
+        cooling_end = convert_date_to_hour(bpr.hvac['cool_ends']) + 23  # end at the last hour of the day
 
         # check if cooling season is at the end of the year (south hemisphere) or in the middle of the year (north)
         if cooling_start < cooling_end and \
```
[the remark this answers is not in the document]
Comment (JIMENOFONSECA) [reply cea/demand/control_heating_cooling_systems.py]: see above

File: cea/demand/datacenter_loads.py [hunk 8]
```diff
@@ -73,27 +73,31 @@ def calc_Qcdataf(locator, bpr, tsd):
...
             # heat pump energy
             tsd['E_cdata'] = np.vectorize(heatpumps.HP_air_air)(tsd['mcpcdata_sys'], (tsd['Tcdata_sys_sup'] + 273),
                                                                 (tsd['Tcdata_sys_re'] + 273), t_source)
             # final to district is zero
             tsd['DC_cdata'] = np.zeros(HOURS_IN_YEAR)
-    elif energy_source == "DC":
-        tsd['DC_cdata'] = tsd['Qcdata_sys']
-        tsd['E_cdata'] = np.zeros(HOURS_IN_YEAR)
-    else:
+        elif energy_source == "NONE":
```
[the remark this answers is not in the document]
Comment (JIMENOFONSECA) [reply cea/demand/datacenter_loads.py]: yes..cooling data. We keep it short since this gets printed out as a column in the demand files

**Comment by JIMENOFONSECA** [comment]
@daren-thomas I solved the next problems:
-> 'keyerror 'code'

**Comment by JIMENOFONSECA** [comment]
creating a scenario works now, but when at it, I clicked in the options of District. and got this messagge. 
image

I think it has to do with the interface

**Comment by JIMENOFONSECA** [comment]
@daren-thomas mind giving me a hand with the fixes in the front-end?

**Comment by daren-thomas** [comment]
I've just pushed changes along with some made to the GUI. Download the [win-unpacked](https://www.dropbox.com/s/hatb0p1i6ym0zks/win-unpacked.zip?dl=0) folder and replace the one in installation dir of CityEnergyAnalyst with the same name to test these changes.

This will need to be merged and released in-sync to make sure it works. I suggest a new (revision) release when merging.

**Pull request merged**